File: src/machine.rs
use crate::enums::{MachineSkill, MachineType};

pub struct Machine {
    pub name: &'static str,
    pub skill: MachineSkill,
    pub machine_type: MachineType,
    pub armored_sides: i32,
    pub weak_sides: i32,
    pub health: i32,
    pub attack: i32,
    pub range: i32,
    pub movement: i32,
    pub points: i32,
}

impl Machine {
    pub fn is_flying(&self) -> bool {
        return self.machine_type == MachineType::Swoop;
    }
}

File: cpp/machine.cpp
#include "machine.h"
#include "enums.h"

bool Machine::is_flying() const
{
    return machine_type == MachineType::Swoop;
}

bool Machine::is_pull() const
{
    return machine_type == MachineType::Pull;
}

Machine::Machine(const char *name, MachineType machine_type, MachineSkill skill, int32_t health, int32_t attack, int32_t range, int32_t movement, MachineSide armored_sides, MachineSide weak_sides, int32_t points)
    : name(name), machine_type(machine_type), skill(skill), health(health), attack(attack), range(range), movement(movement), armored_sides(armored_sides), weak_sides(weak_sides), points(points)
{
}